common/persistence/sql/execution_state_map.go

```diff
@@ -131,6 +131,96 @@ func deleteActivityInfoMap(
 	return nil
 }
 
+func updateUpdateInfoMap(
+	ctx context.Context,
+	tx sqlplugin.Tx,
+	updateInfos map[string]*commonpb.DataBlob,
+	deleteIDs map[string]struct{},
+	shardID int32,
+	namespaceID primitives.UUID,
+	workflowID string,
+	runID primitives.UUID,
+) error {
+
+	if len(updateInfos) > 0 {
+		rows := make([]sqlplugin.UpdateInfoMapsRow, 0, len(updateInfos))
+		for updateID, blob := range updateInfos {
+			rows = append(rows, sqlplugin.UpdateInfoMapsRow{
+				ShardID:      shardID,
+				NamespaceID:  namespaceID,
+				WorkflowID:   workflowID,
+				RunID:        runID,
+				UpdateID:     updateID,
+				Data:         blob.Data,
+				DataEncoding: blob.EncodingType.String(),
+			})
+		}
+
+		if _, err := tx.ReplaceIntoUpdateInfoMaps(ctx, rows); err != nil {
+			return serviceerror.NewUnavailable(fmt.Sprintf("Failed to update update records. Failed to execute update query. Error: %v", err))
```

Comment:
I think it's important to maintain the format shared with with all the other map types (activity info, signal info, &c) so that when this line is eventually grepped out of a log it matches the pattern.